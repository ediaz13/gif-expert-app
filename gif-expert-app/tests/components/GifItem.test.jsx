import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas de <GiftItem /> ', () => {
    
    const title = 'Saitama';
    const url = 'https://one-punchu.com/saitama';

    test('Debe hacer match con el snapshot', () => {
        const { container } = render( <GifItem title={ title } url={ url } /> );

        expect( container ).toMatchSnapshot();
    })

    test('Debe mostrar la imagen con el URL y el ALT indicado', () => {
        render( <GifItem title={ title } url={ url } /> );
        
        //expect( screen.getRole('img').src ).toBe( url );
        //expect( screen.getRole('src').src ).toBe( title );

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
    })

    test('Debe mostrar el titulo del comoponente', () => {
        render( <GifItem title={ title } url={ url } /> );
        expect( screen.getByText( title ) ).toBeTruthy();
        
        
    })
});