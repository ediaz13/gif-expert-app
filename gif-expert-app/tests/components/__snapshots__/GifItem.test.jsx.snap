// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GiftItem />  Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punchu.com/saitama"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
